src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineOption.java

```diff
@@ -22,20 +22,33 @@
 
 import org.eclipse.lsp4j.CompletionItem;
 
-import com.github.cameltooling.lsp.internal.completion.modeline.CamelKTraitManager;
 import com.github.cameltooling.lsp.internal.instancemodel.ILineRangeDefineable;
 
 public class CamelKModelineOption implements ILineRangeDefineable {
 	
 	private String optionName;
-	private String optionValue;
+	private ICamelKModelineOptionValue optionValue;
 	private int startCharacter;
 
 	public CamelKModelineOption(String option, int startCharacter) {
 		int nameValueIndexSeparator = option.indexOf('=');
-		this.optionName = option.substring(0, nameValueIndexSeparator != -1 ? nameValueIndexSeparator : option.length());
-		this.optionValue = nameValueIndexSeparator != -1 ? option.substring(nameValueIndexSeparator+1) : null;
 		this.startCharacter = startCharacter;
+		this.optionName = option.substring(0, nameValueIndexSeparator != -1 ? nameValueIndexSeparator : option.length());
+		this.optionValue = createOptionValue(option, nameValueIndexSeparator);
+	}
+
+	private ICamelKModelineOptionValue createOptionValue(String option, int nameValueIndexSeparator){
+		if(nameValueIndexSeparator != -1) {
+			String value = option.substring(nameValueIndexSeparator+1);
+			int startPosition = getStartPositionInLine() + optionName.length() + 1;
+			if("trait".equals(optionName)) {
```

Comment:
Maybe worth extracting that into a constant.

Reply:
created https://github.com/camel-tooling/camel-language-server/pull/402